examples/linux/io_epoll_test.cpp

```diff
@@ -87,6 +117,115 @@ int main() {
   } catch (const std::exception& ex) {
     std::printf("error: %s\n", ex.what());
   }
+
+  auto pipe_bench = [](auto& rPipeRef, auto& buffer, auto scheduler, int seconds, auto& data, auto& reps, auto& offset) {
+    return defer(
+      [&, scheduler, seconds] {
+        return
+          transform_done(
+            stop_when(
+              // do reads
+              repeat_effect(
+                typed_via(
+                  defer(
+                    [&] {
+                      return transform(
+                        discard(
+                          async_read_some(rPipeRef, as_writable_bytes(span{buffer.data() + 0, 1}))),
+                        [&]{
+                          assert(data[(reps + offset)%sizeof(data)] == buffer[0]);
+                          ++reps;
+                        });
+                    }),
+                  scheduler)),
+              // stop reads after requested time
+              schedule_at(scheduler, now(scheduler) + std::chrono::seconds(seconds))),
```

Comment:
I've been wondering whether we could use the 1-argument `schedule_after(10ms)` here and then rely on "current scheduler" being propagated from a high-level `on()` algorithm wrapping this one.

Of course this would require the scheduler to actually implement `schedule_after(s, d)` though.
But I think we should be able to provide a default implementation of that.